service/src/main/kotlin/fi/espoo/evaka/shared/security/AccessControl.kt

```diff
@@ -145,7 +145,12 @@ WHERE employee_id = :userId
     }
 
     fun requirePermissionFor(user: AuthenticatedUser, action: Action.Global) {
-        assertGlobalPermission(user, action, permittedRoleActions::globalActions)
+        assertPermission(
+            user = user,
+            getAclRoles = { acl.getRolesForAnyUnit(user).roles },
+            action = action,
+            mapping = permittedRoleActions::globalActions
+        )
```

Comment:
The downside here is that the SQL query is run even if the action doesn’t depend on any scoped roles.

Reply:
Actually it seems they aren’t after all 👍

Reply:
Are you sure? It's supposed to be a lambda that is only evaluated when necessary